sale_quotation_capable_vendor_rfq/models/sale_order.py

```diff
@@ -0,0 +1,175 @@
+
+        :param vendor: res.partner
+
+        :return: dict
+        """
+        return {
+            "partner_id": vendor.id,
+            "origin": self.name,
+            "sale_quotation_id": self.id,
+            "currency_id": self.currency_id.id,
+            "company_id": self.company_id.id,
+        }
+
+    def _prepare_rfq_line_vals(self, rfq, line):
+        """
+        Prepare values for purchase.order.line
+
+        :param rfq: purchase.order
+        :param line: dict
+
+        :return: dict
+        """
+        return {
+            "order_id": rfq.id,
+            "product_id": line["line"].product_id.id,
+            "name": line["line"].name,
+            "product_qty": line["line"].product_uom_qty,
+            "product_uom": line["line"].product_uom.id,
+            "price_unit": self._convert_to_currency(
+                line["supplierinfo"].price,
+                line["supplierinfo"].currency_id,
+                rfq.currency_id,
+            ),
+            "date_planned": fields.Datetime.now(),
+        }
+
+    def _get_vendor_dict(self, lead_time=0):
+        """
+        Returns a dict of res.partner records that match the criteria
+
+        :param lead_time: Integer of days to match
+
+        :return: dict
+        """
+        vendors_dict = {}
+        for line in self.order_line:
+            supplierinfo_ids = line.product_id._get_matching_vendor_pricelists(
+                line.product_uom_qty, lead_time
+            )
+            for supplierinfo in supplierinfo_ids:
+                vals = {
+                    "supplierinfo": supplierinfo,
+                    "line": line,
+                }
+                if supplierinfo.partner_id not in vendors_dict:
+                    vendors_dict[supplierinfo.partner_id] = [vals]
+                else:
+                    vendors_dict[supplierinfo.partner_id].append(vals)
+        return vendors_dict
+
+    def _convert_to_currency(self, price, currency_id, currency_to_id):
+        """
+        Convert price to currency
+
+        :param price: float
+        :param currency_id: res.currency
+        :param currency_to_id: res.currency
+
+        :return: float
+        """
+        if currency_id != currency_to_id:
+            price = currency_id._convert(
+                price, currency_to_id, self.env.company, fields.Date.today()
+            )
+        return price
+
+    def _added_alternative_rfq(self):
+        """
+        Added alternative RFQ
+        """
+        self.ensure_one()
+        if len(self.rfq_ids) <= 1:
+            return
+        self.rfq_ids.write(
+            {
+                "purchase_group_id": self._get_alternative_group_id(),
+            }
+        )
+
+    def _get_alternative_group_id(self):
+        """
+        Get alternative group id
+
+        :return: purchase.order.group
+        """
+        self.ensure_one()
+        return self.env["purchase.order.group"].create({}).id
```

Comment:
Method names are confusing.

Do we really need a new method to create empty record?

Also, why word "alternative" is used in this context?